pre-commit.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+function red() {
+	echo "\033[0;31m$@\033[0m"
+}
+
+function green() {
+	echo "\033[0;32m$@\033[0m"
+}
+
+function byellow() {
+	echo "\033[1;33m$@\033[0m"
+}
+
+CHANGED_BY_CARGO_FMT=false
+
+
+echo -e "$(green INFO): Checking code format..."
+
+cargo fmt --all -q -- --check 2>/dev/null
```

Comment:
This command doesn't actually work.  I just tested it by screwing up the format in src/sys/aio.rs, and this command didn't complain.  I think it's caused by this bug: https://github.com/rust-lang/rustfmt/issues/3253 .  That's why we need to do `cargo fmt --all -- --check **/*.rs`